k2/python/k2/mutual_information.py

```diff
@@ -166,10 +170,10 @@ def mutual_information_recursion(
     """
     assert px.ndim == 3
     B, S, T1 = px.shape
-    T = T1 - 1
+    T = py.shape[-1]
+    assert px.shape[-1] in [T, T + 1]  # if T, then "modified".
```

Comment:
https://github.com/pkufool/k2/blob/16f3ac506a725e280016763784713827a993990a/k2/python/k2/mutual_information.py#L94

>       px:
>        A torch.Tensor of some floating point type, with shape ``[B][S][T+1]``,

The doc for `px` should also be updated for the modified case.